verification-by-reference.md

```diff
@@ -0,0 +1,331 @@
+        "architecture": "ppc64le",
+        "os": "linux"
+      }
+    }
+  ]
+}
+```
+
+### web:a2b2 staging verification-config
+
+The fingerprint, or any other verification content, could be placed in the index.config object, leveraging the config object pattern, without depending on loosely defined annotations.
+
+```json
+{
+  "issuedTo": "staging.contoso.com",
+  "issuedBy": "DigCerti SHA2 Extended Validation Server CA",
+  "validFrom": "5/7/2018",
+  "validTo": "6/3/2020",
+  "x509-fingerprint": "43:51:43:a1:b5:fc:8b:b7:0a:3a:a9:b1:0f:66:73:a8"
+}
+```
+
+### web:a2b2 scanning verification w/ index
+
+Another index of type `notary.verification`, which adds the scanning verification.
+
+```json
+{
+  "schemaVersion": 2.1,
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.verification.config.v1+json",
+    "size": 7023,
+    "digest": "sha256:a6a2b2c507a0944348e0303114d8d93aaaa081732b86451d9"
+  },
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 7023,
+      "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451",
+      "platform": {
+        "architecture": "ppc64le",
+        "os": "linux"
+      }
+    }
+  ]
+}
+```
+
+### web:a2b2 scanning verification-config
+
+```json
+{
+  "issuedTo": "*.scan.contoso.com",
+  "issuedBy": "DigCerti SHA2 Extended Validation Server CA",
+  "validFrom": "5/7/2018",
+  "validTo": "6/3/2020",
+  "x509-fingerprint": "86:75:30:90:b5:fc:8b:b7:0a:3a:a9:13:12:55:62:b8"
+}
+```
+
+The nv2 client could retrieve all signatures with the following mocked command
+
+```shell
+nv2 verification pull registry.contoso.com/marketing/web:a2b2
+nv2 verification pull registry.contoso.com/marketing/web@256:efg123
+```
+
+or
+
+```shell
+nv2 verify registry.contoso.com/marketing/web:a2b2
+nv2 verify registry.contoso.com/marketing/web@256:efg123
+```
+
+The `nv2 verify` API compares the artifact with local configuration to attest to it matching the configured rules. These could include the artifact has required verifications for Scanning and Testing verification.
+
+## Pros & Cons
+
+The pros of this approach are:
+
+- Multiple signatures may be added without changing the original artifact digest as no content changes are made to the digest, or even annotations
+- All deployment references remain: `web:a2b2` and it’s associated digest do not change.
+
+The cons of this approach:
+
+- A new API must be added. We’ve said this was within scope and ok as registries must add Notary v1, Content Trust to support existing scenarios. Since most registries need to do some work, adding a new API is not a stretch
+- Reverse lookup, and its impact on garbage collection. Registries tend to traverse down. If multiple manifests reference common base layers, the registry knows to reference count shared layers and only deletes the layers once all manifests are removed. Orphaned manifests may be deleted if they’re no longer referenced by tags.  
```

Comment:
> Orphaned manifests may be deleted if they’re no longer referenced by tags. 

I don't really agree with this, in general. Images are addressable by digest, so it's not always safe to just GC orphaned manifests. It's a reasonable policy to enable, but this would be a dangerous default.

Reply:
Deleting untagged manifests is a choice of a registry operator and a customer. We'll have more info on this as we've had a few discussion if cross references should be tracked by manifests and indexes? 
Or, is there an additional way to track references?
- Reading a config object?
- A separate API to link a signature to a manifest? Having a separate API would enable a different role for SIGNER. But, we could also say the role should be applied to a given Artifact type: `vnd.cncf.notary.config.v2` for instance.

I think we agree on the constraints of the pros/cons and need to figure out a tenable solution.